profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -247,8 +277,9 @@ export function UnnecessaryEffectTriggering() {
 
   useEffect(() => {
     const interval = setInterval(async () => {
+      // next line is returning a new object every time,
       const leader = await fetchLeader()
-      setLeader(leader)
+      setLeader(prev => leader.name !== prev.name ? leader : prev)
```

Comment:
Does react recognize there is no state changes here? I think this probably triggers a re-render anyway.

Reply:
I had the same doubt but seems like the param shares the same memory address.